www_src/pages/section-2/section-2.jsx

```diff
@@ -170,19 +212,19 @@ var Grid = React.createClass({
     this.gridWidth = undefined;
     this.gridHeight = undefined;
 
+    this.slotWidth = undefined;
+    this.slotHeight = undefined;
+
     var nodes = [];
     var layout = this.state.layout;
 
-    var tilesPerRow = layout[0].length;
-    var tilesPerCol = layout.length;
+    this.tilesPerRow = layout[0].length;
+    this.tilesPerCol = layout.length;
 
     // Parse aspect ratio
     var widthAR = parseInt(this.props.aspectRatio.split(':')[0], 10);
     var heightAR = parseInt(this.props.aspectRatio.split(':')[1], 10);
 
-    var slotWidth;
-    var slotHeight;
-
     // Determine if a slot has a neighboring page in any cardinal direction
     function hasNeighbors(x, y) {
```

Comment:
should this function be redeclared each time `render` runs? It might make sense to make this a `this.hasNeighbours`